public/app/plugins/panel/xychart/XYChartPanel.tsx

```diff
@@ -215,47 +217,68 @@ export const XYChartPanel2 = (props: Props) => {
     <>
       <VizLayout width={props.width} height={props.height} legend={renderLegend()}>
         {(vizWidth: number, vizHeight: number) => (
-          <UPlotChart config={builder} data={facets} width={vizWidth} height={vizHeight} />
+          <UPlotChart config={builder} data={facets} width={vizWidth} height={vizHeight}>
+            {config.featureToggles.newVizTooltips && props.options.tooltip.mode !== TooltipDisplayMode.None && (
```

Comment:
Should we also wrap the `UPlotChart` in the FF check / options check? does it make sense to render empty UPlotChart here?

Reply:
UPlotChart is the only component we are using to render xy chart, that's why it's outside the FF. Old xy chart tooltip exists outside xy chart.